### PR TITLE
auth: add scopes on token request and a parameter customize them

### PR DESCRIPTION
**Pull request by jagobagascon**
The requested scopes can be customised. When unset, the default scopes will be requested: `openid`, `offline`, `https://saltoapis.com/auth/nebula`.

### CODE REVIEW

File: src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs [hunk 1]
```diff
@@ -92,15 +95,23 @@ class SaltoOAuthClient : OAuthClientCredentialsProvider
...
         SaltoTokenResponse token;
 
-        readonly String clientId;
-        readonly String clientSecret;
+        readonly String   clientId;
+        readonly String   clientSecret;
+        readonly String[] scopes;
+
+        public static readonly String[] DefaultScopes = new String[] {
+            "openid", "offline", "https://saltoapis.com/auth/nebula"
```
Comment (jagobagascon) [inline src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs:103]: @glerchundi when the user does not specify any scope we are requesting these by default, is this ok? Or should we force them to specify scopes they want?

File: src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs [hunk 2]
```diff
@@ -187,7 +201,7 @@ async Task<SaltoTokenResponse> ObtainOAuthToken(HttpClient httpClient)
                 {
                     // get error
                     var error = JsonSerializer.Deserialize<TokenError>(result);
-                    throw new SaltoOAuthException($"Error getting the access_token. Server returned {responseMessage.StatusCode} error ( error = '{error.Error}', description = '{error.ErrorDescription}')");
+                    throw new SaltoOAuthException($"Error getting the access_token. Server returned {responseMessage.StatusCode} error ( error = '{error.Error}', error_hin = '{error.ErrorHint}', description = '{error.ErrorDescription}')");
```
Comment (glerchundi) [inline src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs:204]: Typo `error_hin`.

File: src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs (hunk 1 above)
Comment (glerchundi) [reply src/Saltoapis.Auth/SaltoapisClientCredentialsProvider.cs:103]: I would say that we can start without defining this default scopes.

**Review by glerchundi — APPROVED** [review]
Get rid of the default scopes and we'll see how to proceed afterwards. Take into consideration that each service could have different scope so these defaults will get outdated sooner than we expect.

**Pull request merged**